### DISCUSSION TITLE
How do I setup unit tests with code coverage for my monorepo that contains nodejs and dotnet apps?

### DISCUSSION THREAD

**Discussion by craigbroadman** [nx-dotnet/nx-dotnet#843] [General]
I use the following command as part of a GitHub Actions workflow to execute unit tests with code coverage metrics:

`npx nx affected -t test --parallel=3 --ci --code-coverage --base=origin/main --head=HEAD`

To date, all of our apps are nodejs using jest with the following `jest.config.ts` file:

```ts
/* eslint-disable */
export default {
  displayName: 'my-app',
  preset: '../../../jest.preset.js',
  globals: {},
  setupFiles: ['../../../jest.setup.ts'],
  testEnvironment: 'node',
  transform: {
    '^.+\\.[tj]s$': [
      'ts-jest',
      {
        tsconfig: '<rootDir>/tsconfig.spec.json',
      },
    ],
  },
  moduleFileExtensions: ['ts', 'js', 'html'],
  coverageDirectory: '../../../output/apps/my-app',
  coverageReporters: ['text', 'cobertura'],
  testResultsProcessor: 'jest-junit',
  reporters: [
    'default',
    [
      'jest-junit',
      {
        outputDirectory: './output/apps/my-app',
        outputName: 'unit-test-results.xml',
      },
    ],
  ],
};
```

However, now that I've introduced a dotnet app, the command fails with `Unknown switch` due to the `--ci` and `--code-coverage`

Does anyone have any advice on what to change to get this working? I'm hoping to avoid having to have separate commands for `test` and `test-with-coverage`?

**Comment by AgentEnder** [comment]
I'd recommend moving all of the flags into a target configuration and running with something akin to `--configuration ci`

**Reply by craigbroadman** [reply]
I like that idea @AgentEnder ✅ 

I assume the configurations need to go into the project.json file, within the "test" target, something like the below:

```json
"test": {
      "executor": "@nx-dotnet/core:test",
      "cache": false,
      "dependsOn": ["build"],
      "options": {
        "noBuild": true
      },
      "configurations": {
        "my-name": {
          "values-here?": true
        }
      }
    }
```

Could I ask what the syntax should be to be able to pass through commands below to dotnet test, for example: 

```
/p:CollectCoverage=true 
/p:CoverletOutput=..\code-coverage\coverage 
/p:MergeWith=..\code-coverage\coverage.json 
/p:CoverletOutputFormat=json 
/p:ExcludeByFile=\"**\Connected Services\ThirdPartyReceiver\Reference.cs\"
```

**Reply by AgentEnder** [reply]
If theres not a direct arg for something, you can use extraParameters and pass them via string: https://www.nx-dotnet.com/docs/core/Executors/test#extraparameters
